### ISSUE TITLE
Collect solo files into a single, separate output group

### ISSUE THREAD

**Issue by dhruvkb** [pls-rs/pls#82]
## Description

Solo files should be collected into a single table/grid when listing. If separated into their own groups, they waste a lot of space.

```console
$ pls /bin/ls /usr/bin/sqlite3 /Users/dhruvkb/Developer/personal/mega/pls -g=true
```

<img width="643" alt="Screenshot 2023-08-28 at 6 21 35 AM" src="https://github.com/dhruvkb/pls/assets/16580576/2a28bcfe-c079-45d6-963a-b7426f426aac">

## Expectations

Here's how `ls` and `exa` both handle the same input arguments.

<img width="611" alt="Screenshot 2023-08-28 at 6 22 09 AM" src="https://github.com/dhruvkb/pls/assets/16580576/9322462b-1db3-49d5-9747-c3f443ca9118">

The individual files are collected and displayed as one grid, and the directory appears in its own grid with a heading.

## Workarounds

In certain specific cases, such as when listing files in the same directory, name-filters can be used to emulate the expected behaviour.

<img width="266" alt="Screenshot 2023-08-28 at 6 28 52 AM" src="https://github.com/dhruvkb/pls/assets/16580576/d30ca2be-144c-495a-b8a0-6d3cdf621f3b">

## Additional information

This issue was raised in discussion #81.

**Comment by dhruvkb** [comment]
I spent quite some time working on this issue this weekend. It turns out the problem is a little more complicated than just collecting all solo files into a group and rendering them together.

One important issue is because two files may be in different locations, they can contain different, conflicting values in common configurations. For example, if two files have different `.pls.yml` configs that define different values for `constants.table.column_names`, what should be the actual heading when these are combined together?

**Comment by joaociocca** [comment]
one way I think you tackle that issue you raised would be to separate files by location.
for example, a `pls *.json`  will currently do this:

![image](https://github.com/pls-rs/pls/assets/6473725/97d3ac30-710e-4e24-9f98-b669d6de1607)

despite all the json files being in the same directory, each are getting their own header, specifically in the format of:

```
<filename>
<header>
<details>
```

one way to tackle this would be if we'd have:

```
<location>
<header>
<loop of files details> 
```

kinda like `ls` itself does:
![image](https://github.com/pls-rs/pls/assets/6473725/acab7227-fc18-42f0-8596-02abdd0315f0)

for the multiple location problem, you could either 1) separate by location at the location before header (I think this would fit pls better) or 2) include the location in the filename at the details (again, like `ls` itself does):

![image](https://github.com/pls-rs/pls/assets/6473725/1203b1fc-f40a-44f9-a922-acf9bbcfff34)

I had to redact there, but it's a simple directory name.

**Comment by dhruvkb** [comment]
Here to leave a quick status update: I just pushed some changes that completely overhauled most of the app to make this happen. After making sure everything works and will create a new release shortly.

**Comment by dhruvkb** [comment]
I just released [v0.0.1-beta.3](https://github.com/pls-rs/pls/releases/tag/v0.0.1-beta.3) which contains [this feature](https://pls-rs.github.io/pls/guides/paths/#solo-files).

**Comment by joaociocca** [comment]
> I just released [v0.0.1-beta.3](https://github.com/pls-rs/pls/releases/tag/v0.0.1-beta.3) which contains [this feature](https://pls-rs.github.io/pls/guides/paths/#solo-files).

coming back after a while =) I think you need to recheck the link :eyes: 

https://pls.cli.rs/guides/paths/#solo-files works, but https://pls-rs.github.io/pls/guides/paths/#solo-files gives 404
